Content.Server/Atmos/GasMixture.cs

```diff
@@ -0,0 +1,413 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Content.Shared.Atmos;
+using Logger = Robust.Shared.Log.Logger;
+using Math = CannyFastMath.Math;
+using MathF = CannyFastMath.MathF;
+
+namespace Content.Server.Atmos
+{
+    /// <summary>
+    ///     A general-purposes, variable volume gas mixture.
+    /// </summary>
+    public class GasMixture
+    {
+        private readonly float[] _moles = new float[Atmospherics.TotalNumberOfGases];
+        private readonly float[] _molesArchived = new float[Atmospherics.TotalNumberOfGases];
```

Comment:
I am fairly certain from previous discussions I've had about TG atmos that this moles-archived thing is kinda a silly wart of their system that can be implemented more efficiently (iirc it's ONLY used during tile atmos and that means it would be better served stored there instead of *every* gas mixture). I'm not experienced enough with the code to say for sure though.

Reply:
I can't really say for sure either, and seems like temperature exchange and other 'critical' code depends on this being there so... I'm gonna leave it like this until I figure it out.